pylint/checkers/dataclass_checker.py

```diff
@@ -0,0 +1,137 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Dataclass checkers for Python code."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+def _is_dataclasses_module(node: nodes.NodeNG) -> bool:
+    """Utility function to check if node is from dataclasses_module."""
+    return isinstance(node, nodes.Module) and node.name == "dataclasses"
```

Comment:
Sadly there are other types of `dataclasses` as well. There is actually a list that we maintain in `astroid`. You could try importing it:
https://github.com/PyCQA/astroid/blob/main/astroid/brain/brain_dataclasses.py